src/node_contextify.cc

```diff
@@ -392,23 +419,24 @@ void ContextifyContext::WeakCallback(
 ContextifyContext* ContextifyContext::ContextFromContextifiedSandbox(
     Environment* env,
     const Local<Object>& sandbox) {
-  MaybeLocal<Value> maybe_value =
-      sandbox->GetPrivate(env->context(),
-                          env->contextify_context_private_symbol());
-  Local<Value> context_external_v;
-  if (maybe_value.ToLocal(&context_external_v) &&
-      context_external_v->IsExternal()) {
-    Local<External> context_external = context_external_v.As<External>();
-    return static_cast<ContextifyContext*>(context_external->Value());
+  Local<Value> context_global;
+  if (sandbox
+          ->GetPrivate(env->context(), env->contextify_context_private_symbol())
+          .ToLocal(&context_global) &&
+      context_global->IsObject()) {
+    return Unwrap<ContextifyContext>(context_global.As<Object>());
   }
   return nullptr;
 }
 
-// static
 template <typename T>
 ContextifyContext* ContextifyContext::Get(const PropertyCallbackInfo<T>& args) {
+  return Get(args.This());
+}
+
+ContextifyContext* ContextifyContext::Get(Local<Object> object) {
   Local<Context> context;
-  if (!args.This()->GetCreationContext().ToLocal(&context)) {
+  if (!object->GetCreationContext().ToLocal(&context)) {
```

Comment:
I tried using a private symbol from the receiver to the wrapper and then unwrap to get to the ContextifyContext*, which is a bit slower than the current approach, so we are still using the context embedder field here. In any case, as long as the context is alive, the sandbox must be alive (through the kSandboxObject internal field), then the wrapper must be alive (through the contextify_context_private_symbol), so the `ContextifyContext` is also alive. On the other hand as long as the `ContextifyContext` is alive, the wrapper is alive, then the context must also be alive (through the constructor).